subsys/net/ip/Kconfig

```diff
@@ -174,6 +174,156 @@ config NET_SHELL
 	  Activate shell module that provides network commands like
 	  ping to the console.
 
+config NET_SHELL_SHOW_DISABLED_COMMANDS
```

Comment:
Wrap the entire block in

```
if NET_SHELL
# ...
endif # NET_SHELL
```

To avoid all the `depends on NET_SHELL` entries.